### ISSUE TITLE
 ERROR:  duplicate key value violates unique constraint "status_pkey"

### ISSUE THREAD

**Issue by elandorr** [mattermost/docker#21]
After upgrading infinite of those errors show up, in addition to the previously occurring errors which the mattermost team said are 'normal' on the forum.
```
STATEMENT:  insert into "status" ("userid","status","manual","lastactivityat","dndendtime","prevstatus") values ($1,$2,$3,$4,$5,$6);
ERROR:  duplicate key value violates unique constraint "status_pkey"
```
While I'm at it, here's another error:

```
error   app/server.go:527       Mail server connection test is failed    {"error": "SendEmailNotifications is not true", "errorVerbose": "SendEmailNotifications is not true\ngith
ub.com/mattermost/mattermost-server/v5/shared/mail.TestConnection\n\tgithub.com/mattermost/mattermost-server/v5/shared/mail/mail.go:212\ngithub.com/mattermost/mattermost-server/v5/app.NewServer\n\tgithub.com/mattermost/matterm
ost-server/v5/app/server.go:526\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.runServer\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:75\ngithub.com/mattermost/mattermost
-server/v5/cmd/mattermost/commands.serverCmdF\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:59\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:852\ngithu
b.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/mattermost/mattermost-server/v5/cmd/m
attermost/commands.Run\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/root.go:14\nmain.main\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/main.go:31\nruntime.main\n\truntime/proc.go:204\nr
untime.goexit\n\truntime/asm_amd64.s:1374"}       
```

_But wait, there's more!_

```
termost-db | 2021-08-12 16:38:41.775 UTC [30] ERROR:  constraint "fk_retentionpolicieschannels_retentionpolici
es" for relation "retentionpolicieschannels" already exists
mattermost-db | 2021-08-12 16:38:41.775 UTC [30] STATEMENT:
mattermost-db |                 ALTER TABLE RetentionPoliciesChannels
mattermost-db |                 ADD CONSTRAINT FK_RetentionPoliciesChannels_RetentionPolicies
mattermost-db |                 FOREIGN KEY (PolicyId) REFERENCES RetentionPolicies (Id)
mattermost-db |                 ON DELETE CASCADE;                                                               mattermost-db | 2021-08-12 16:38:41.775 UTC [30] ERROR:  constraint "fk_retentionpoliciesteams_retentionpolicies$
 for relation "retentionpoliciesteams" already exists                                                            mattermost-db | 2021-08-12 16:38:41.775 UTC [30] STATEMENT:
mattermost-db |                 ALTER TABLE RetentionPoliciesTeams                                               mattermost-db |                 ADD CONSTRAINT FK_RetentionPoliciesTeams_RetentionPolicies
mattermost-db |                 FOREIGN KEY (PolicyId) REFERENCES RetentionPolicies (Id)                         mattermost-db |                 ON DELETE CASCADE;
```

Could someone finally deal with this please?
There is no point to having an error log if it spams thousands of errors. On the forum you previously said 'it's normal, ignore' and I did so, but now the new error literally spams dozens of times per second.

Fellow upgraders also note that the port changed from `8000 ` to `8065`, you have to edit that in the mm config file.

For the record: previous version was 5.37 via the old method, new one via the prebuilt images is 5.37.1.

**Comment by elandorr** [comment]
Way too many errors to even dig though, and they sound quite severe:

```
runtime error: invalid memory address or nil pointer dereference
goroutine 6950 [running]:
net/http.(*conn).serve.func1(0xc0054bc820)
net/http/server.go:1801 +0x147
panic(0x2521b60, 0x3f2d9d0)
```

Has nobody ever tested this before publishing? How can nobody have looked at the logs first?

This setup is wholly standardized exactly as per your instructions, no plugins, nothing extraneous.

**Comment by mrckndt** [comment]
```
STATEMENT:  insert into "status" ("userid","status","manual","lastactivityat","dndendtime","prevstatus") values ($1,$2,$3,$4,$5,$6);
ERROR:  duplicate key value violates unique constraint "status_pkey"
```
and
```
termost-db | 2021-08-12 16:38:41.775 UTC [30] ERROR:  constraint "fk_retentionpolicieschannels_retentionpolici
es" for relation "retentionpolicieschannels" already exists
mattermost-db | 2021-08-12 16:38:41.775 UTC [30] STATEMENT:
mattermost-db |                 ALTER TABLE RetentionPoliciesChannels
mattermost-db |                 ADD CONSTRAINT FK_RetentionPoliciesChannels_RetentionPolicies
mattermost-db |                 FOREIGN KEY (PolicyId) REFERENCES RetentionPolicies (Id)
mattermost-db |                 ON DELETE CASCADE;                                                               mattermost-db | 2021-08-12 16:38:41.775 UTC [30] ERROR:  constraint "fk_retentionpoliciesteams_retentionpolicies$
 for relation "retentionpoliciesteams" already exists                                                            mattermost-db | 2021-08-12 16:38:41.775 UTC [30] STATEMENT:
mattermost-db |                 ALTER TABLE RetentionPoliciesTeams                                               mattermost-db |                 ADD CONSTRAINT FK_RetentionPoliciesTeams_RetentionPolicies
mattermost-db |                 FOREIGN KEY (PolicyId) REFERENCES RetentionPolicies (Id)                         mattermost-db |                 ON DELETE CASCADE;
```
are harmless. Sometimes statements are erroring out intentionally and then proceed accordingly.

```
error   app/server.go:527       Mail server connection test is failed    {"error": "SendEmailNotifications is not true", "errorVerbose": "SendEmailNotifications is not true\ngith
ub.com/mattermost/mattermost-server/v5/shared/mail.TestConnection\n\tgithub.com/mattermost/mattermost-server/v5/shared/mail/mail.go:212\ngithub.com/mattermost/mattermost-server/v5/app.NewServer\n\tgithub.com/mattermost/matterm
ost-server/v5/app/server.go:526\ngithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.runServer\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:75\ngithub.com/mattermost/mattermost
-server/v5/cmd/mattermost/commands.serverCmdF\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/server.go:59\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:852\ngithu
b.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/mattermost/mattermost-server/v5/cmd/m
attermost/commands.Run\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/root.go:14\nmain.main\n\tgithub.com/mattermost/mattermost-server/v5/cmd/mattermost/main.go:31\nruntime.main\n\truntime/proc.go:204\nr
untime.goexit\n\truntime/asm_amd64.s:1374"}
```
arises because there is no SMTP server configured (or it is failing).

> Fellow upgraders also note that the port changed from 8000 to 8065, you have to edit that in the mm config file.

Mattermost is listening on 8065 since forever and the previous community approach used the custom port 8000/

**Comment by mrckndt** [comment]
> Way too many errors to even dig though, and they sound quite severe:
> 
> ```
> runtime error: invalid memory address or nil pointer dereference
> goroutine 6950 [running]:
> net/http.(*conn).serve.func1(0xc0054bc820)
> net/http/server.go:1801 +0x147
> panic(0x2521b60, 0x3f2d9d0)
> ```
> 
> Has nobody ever tested this before publishing? How can nobody have looked at the logs first?
> 
> This setup is wholly standardized exactly as per your instructions, no plugins, nothing extraneous.

With 6 lines of stack trace without any logs around i can't say what is failing. This can be any plugin or something inside Mattermost. Please share more of the log.

**Comment by tuxity** [comment]
> ```
> STATEMENT:  insert into "status" ("userid","status","manual","lastactivityat","dndendtime","prevstatus") values ($1,$2,$3,$4,$5,$6);
> ERROR:  duplicate key value violates unique constraint "status_pkey"
> ```

I saw an issue for that one, [MM-37202](https://mattermost.atlassian.net/browse/MM-37202), should be fixed soon I guess.

> Fellow upgraders also note that the port changed from 8000 to 8065, you have to edit that in the mm config file.

I had to change it too, no big deal. Maybe we can add a note in Readme to remind the people upgrading from community ?

> ```
> runtime error: invalid memory address or nil pointer dereference
> goroutine 6950 [running]:
> net/http.(*conn).serve.func1(0xc0054bc820)
> net/http/server.go:1801 +0x147
> panic(0x2521b60, 0x3f2d9d0)
> ```

I saw this one too after playing with plugins, It was after enabling Focalboard or Incident Collaboration I can try to reproduce with more logs.

**Comment by mrckndt** [comment]
> > ```
> > STATEMENT:  insert into "status" ("userid","status","manual","lastactivityat","dndendtime","prevstatus") values ($1,$2,$3,$4,$5,$6);
> > ERROR:  duplicate key value violates unique constraint "status_pkey"
> > ```
> 
> I saw an issue for that one, [MM-37202](https://mattermost.atlassian.net/browse/MM-37202), should be fixed soon I guess.
> 
> > Fellow upgraders also note that the port changed from 8000 to 8065, you have to edit that in the mm config file.
> 
> I had to change it too, no big deal. Maybe we can add a note in Readme to remind the people upgrading from community ?

Yes, sure. The repo isn't fully finished yet and needs some improvements/scripts for sure. I'm working on it in parallel to many other things so please be patient. Other than that we're always happy about PRs. :)

> 
> > ```
> > runtime error: invalid memory address or nil pointer dereference
> > goroutine 6950 [running]:
> > net/http.(*conn).serve.func1(0xc0054bc820)
> > net/http/server.go:1801 +0x147
> > panic(0x2521b60, 0x3f2d9d0)
> > ```
> 
> I saw this one too after playing with plugins, It was after enabling Focalboard or Incident Collaboration I can try to reproduce with more logs.

**Comment by elandorr** [comment]
> With 6 lines of stack trace without any logs around i can't say what is failing. This can be any plugin or something inside Mattermost. Please share more of the log.

Your logs do not show anything else - this is literally all.

And as I said, there are zero plugins enabled, not even the included bots.

Tell me what you need to fix and I'll try to provide it.

**Comment by elandorr** [comment]
>please be patient.

It took you more than an entire year to even address the gaping issue from the previous repo. I'd call that plenty patient.

> Other than that we're always happy about PRs. :)

Sure, if you were true open source I'd be happy to dig into it as my unpaid free time allows, but as basics such as maintenance (no retention policy/infinitely exploding storage requirements) or permissions (users can kick anyone on the server, including admins, reported that in another issue) are not even there, the team edition is already very obviously nothing but a marketing gag to promote the commercial offering.

I understand things take time, but to free up such time you could simply stop releasing new things nobody asked for every 5 minutes, and just focus on what matters: uptime/maintenance/stability. It's getting ridiculous to have new stuff to deal with every week yet basics are unfixed for years.

**Comment by tuxity** [comment]
Easy @elandorr, you are in your right to think like this but, I think it's not the right place to express your concerns :/

**Comment by elandorr** [comment]
I replied to a statement given. Don't post a statement if you don't want a reply to it.

On topic:

Would it be helpful to use the bulk export tool https://docs.mattermost.com/manage/bulk-export-tool.html, set up an entirely new server, changing the configuration manually, and then importing users/posts/teams again?

**Comment by emilyacook** [comment]
Hello @elandorr, regarding this error

```
STATEMENT:  insert into "status" ("userid","status","manual","lastactivityat","dndendtime","prevstatus") values ($1,$2,$3,$4,$5,$6);
ERROR:  duplicate key value violates unique constraint "status_pkey"
```
As @tuxity pointed out, the fix for this was backported to 5.37 so it's scheduled to be included in v5.37.2: https://mattermost.atlassian.net/browse/MM-37202

> Would it be helpful to use the bulk export tool https://docs.mattermost.com/manage/bulk-export-tool.html, set up an entirely new server, change the configuration manually, and then import users/posts/teams again?

I don't think I would recommend that unless you're still having severe disruption after v5.37.2 and trying the other things discussed in this thread. Since the error is on the database side it might not circumvent the issue.

I understand how this is frustrating, but for some context the old repository was community maintained and not an official Mattermost supported project. The migration effort is still a work in progress, so not everything is fully tested or documented yet. I appreciate you opening this issue though since we definitely don't want people to be frustrated, so it's good to know when someone is. We'll make sure to get some of this added to our guides at least as well as work with QA to see if there are some testing blindspots we can work out.

**Comment by mrckndt** [comment]
Closing this as the initial error is intentionally and used for error handling.
